doc/developer/design/20230519_statement_logging.md

```diff
@@ -0,0 +1,232 @@
+| `finished_at`           | `timestamp with time zone` | The time at which the statement finished, or null if it is still running.                                |
+| `was_successful`        | `bool`                     | Whether the statement terminated successfully, or null if it is still running.                           |
+| `was_canceled`          | `bool`                     | Whether the statement was canceled, or null if it is still running.                                      |
+| `error_message`         | `text`                     | The error message. Non-null if and only if the statement finished without succeeding or being cancelled. |
+| `rows_returned`         | `bigint`                   | How many rows the statement returned, or null if it is still running, errored or was canceled.           |
+| `was_fast_path`         | `bool`                     | Whether the statement executed on the fast path, or null if it is still running.                         |
+
+
+**mz_session_history**
+
+| Field              | Type   | Meaning                                               |
+|--------------------|--------|-------------------------------------------------------|
+| `id`               | `uuid` | The unique identifier of the session.                 |
+| `application_name` | `text` | The name of the application that created the session. |
+| `user_name`        | `text` | The name of the user who owns the session.            |
+
+
+### Observability and Rollout
+
+If we misjudge the size and volume of our typical users' queries, we
+will log much more data than we expect. Thus, rollout should proceed
+in two phases. First, we should enable the feature in "dry run" mode,
+which does not actually log any queries, but tracks (as a Prometheus
+metric) how much data would have been logged had the feature actually
+been turned on. Only once we validate that it meets expectations
+should we actually enable it. We should do something similar before
+adding any new columns to the tables (e.g., the plans).
+
+### Testing
+
+For testing, we will probably need a way to make the random sampling
+deterministic; this can be accomplished by adding a
+`statement_log_random_seed` system var that we set from within
+testdrive or SLTs.
+
+## Alternatives
+
+Log all executed statements, rather than sampling. This approach was
+rejected because users might come to rely on all statements being logged,
+which would require us making all statements durable, dramatically
+increasing the cost of high-QPS usage patterns. Instead, we will not
+allow setting the sampling rate above some value slightly less than 1,
+e.g. 0.99.
+
+## Open questions
+
+* Should we let the user set their own sampling rate, or does this
+  have the potential to put unacceptable load on e.g. Cockroach?
+* What should be the default values of the system vars?
```

Comment:
Is it accurate to assume that the default value of `statement_log_max_age` will be 30 days?